variables.tf

```
@@ -1,3 +1,8 @@
+variable "authorized_networks" {
+  type = map(string)
```

Maybe use `list(object{})` instead to support optional `expiration_time`?
```suggestion
  type = list(object({
    name = optional(string) # Not sure about optional. It's optional for the API, we might want to enforce it
    value = string
    expiration_time = optional(string)
  })
```
Alternatively:
```suggestion
  type = map(object({ # map key is CIDR
    name = optional(string) # Not sure about optional. It's optional for the API, we might want to enforce it
    expiration_time = optional(string)
  })
```

---

To be clear, I'm okay with leaving it as it is. Just considering flexibility (or over engineering)

---

I actually had it using an `object` initially but changed into into something "simpler". Didn't think about the possibility of (future) additional parameters. Going back to `object` it is.

---

Benefit is the first approach is user readability, benefit of the second approach is stripping the need to create a custom for loop (though that's exactly the kind of thing we want to stow away in a module).

---

Will postpone supporting `expiration_time` until we actually need it.

---

Regarding the first proposal, maybe use `cidr` instead of `value` for a key, as it's more descriptive.